README.md

```diff
@@ -50,6 +50,28 @@ have dependencies on other algebras which must be implemented.
     - Two promises are equivalent when they yield equivalent values.
     - Two functions are equivalent if they yield equivalent outputs for equivalent inputs.
 
+## Type signature notation
+
+The type signature notation used in this document is described [here](https://sanctuary.js.org/#types). In summary:
+
+1. New types can be created via type constructors. Type constructors take one or more type arguments. For example, the type "array of strings" is written as `Array String`, and the type constructor `Array` can be defined as `Array :: a -> a`.
```

Comment:
These `ghci` (Haskell) examples may be instructive:

```haskell
> :kind String
String :: *

> :kind Maybe
Maybe :: * -> *

> :kind Maybe String
Maybe String :: *
```

One can think of `*` as the type of types. The type of `String` is `*` – `String` is a type. The type of `Maybe` is `* -> *` – `Maybe` is a type *constructor*.